recipes/caffe/build.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+# Setup CMake build location
+mkdir build
+cd build
+
+# Configure, build, test, and install.
+cmake -DCPU_ONLY=1 -DBLAS="open" -DCMAKE_INSTALL_PREFIX="${PREFIX}" ..
+make
+make runtest
+make install
+
+# Python installation is non-standard. So, we're fixing it.
+mv "${PREFIX}/python/caffe" "${SP_DIR}/"
+find "${PREFIX}/python/" -name "*.py" | xargs chmod +x
+find "${PREFIX}/python/" -name "*.py" | xargs -I {} cp {} "${PREFIX}/bin/"
```

Comment:
> Also, what do you mean by "intended to be binaries"?

Yeah, this is fun too. They state at the top of these Python files that they are meant for use as CLIs. Take [`classify.py`]( https://github.com/BVLC/caffe/blob/master/python/classify.py#L1-L3 ), for instance, it has the logic to run under Python on Mac and Linux from the command line and has a note that this is the intent. The rest are the same (though feel free to check). All of the files lack executable permissions, have `.py` extensions, lack any real Python callable API (except for `main`), and are not placed in the usual `bin` directory.

> It is unusual to have python files directly executable like that, but I can remember doing such things as a scientist learning to code.

Sure, when I was still learning Python, I did a few things like this. To be honest, they should be lauded for making a Python interface to Caffe and, to be honest, the Python code that I have seen (though haven't seen all of it) looked pretty clean for the most part. It is just their package management side that is a bit weak. Then again this is a weakness of Python too and a bit rough to follow too. So maybe they just thought this would be the easiest way to get it out there.